test/integration/admin/api/services/proxy/policies_controller_test.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class PoliciesControllerTest < ActionDispatch::IntegrationTest
+
+  def setup
+    @provider = FactoryBot.create(:provider_account)
+    @service = @provider.default_service
+    host! @provider.external_admin_domain
+    @access_token = FactoryBot.create(:access_token, owner: @provider.admin_users.first!, scopes: %w[account_management]).value
+  end
+
+  class UpdateTest < PoliciesControllerTest
+    test 'prints chain level errors' do
+      valid_config = { name: 'apicast', configuration: {}, version: 'builtin', enabled: true, removable: false }
+
+      Proxy::PoliciesConfig.stub_const(:MAX_LENGTH, 16.bytes) do
+        put admin_api_service_proxy_policies_path(@service, access_token: @access_token, format: :json), params: {
+          proxy: { policies_config: valid_config }
+        }, as: :json
+      end
+      resp_body = JSON.parse(response.body)
+
+      assert_response :unprocessable_entity
+      assert_equal 'is too long (maximum is 16 characters)', resp_body['errors']['policies_config'][0]
+    end
+
+    test 'prints policy level errors' do
+      invalid_config = { name: 'apicast', configuration: {}, enabled: true, removable: false }
+
+      put admin_api_service_proxy_policies_path(@service, access_token: @access_token, format: :json), params: {
+        proxy: { policies_config: invalid_config}
+      }, as: :json
+      resp_body = JSON.parse(response.body)
+
+      assert_response :unprocessable_entity
+      assert_equal "can't be blank", resp_body['policies_config'][0]['errors']['version'][0]
+    end
+
+    test 'policy level errors are promoted to chain errors as well' do
+      invalid_config = { name: 'apicast', configuration: {}, enabled: true, removable: false }
+
+      Proxy::PoliciesConfig.stub_const(:MAX_LENGTH, 16.bytes) do
+        put admin_api_service_proxy_policies_path(@service, access_token: @access_token, format: :json), params: {
+          proxy: { policies_config: invalid_config}
+        }, as: :json
+      end
+      resp_body = JSON.parse(response.body)
+
+      assert_response :unprocessable_entity
+      assert_equal 'contains some invalid policy', resp_body['errors']['policies_config'][0]
```

Comment:
In this case there will be two errors, I think the test should reflect this (and the "invalid policy" one is not necessarily on the 0 position (I mean, even if it is because of the explicit order in the code, it is not important here).

I'd rather do here:
```
assert_same_elements ['contains some invalid policy', 'is too long (maximum is 16 characters)'], resp_body['errors']['policies_config']
```

Reply:
I didn't know about `assert_same_elements`. I pushed the changes, thanks!